modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNodeClientCodegen.java

```diff
@@ -141,9 +243,54 @@ private void addNpmPackageGeneration() {
         supportingFiles.add(new SupportingFile("package.mustache", getPackageRootDirectory(), "package.json"));
         supportingFiles.add(new SupportingFile("tsconfig.mustache", getPackageRootDirectory(), "tsconfig.json"));
     }
+    
+    private String getIndexDirectory() {
+        String indexPackage = modelPackage.substring(0, Math.max(0, modelPackage.lastIndexOf('.')));
+        return indexPackage.replace('.', File.separatorChar);
+    }
+
+    @Override
+    public String getSchemaType(Schema p) {
+        String openAPIType = super.getSchemaType(p);
```

Comment:
could you please explain in the comments why these type mapping methods are needed / what they do?

Reply:
@macjohnny I believe the idea for this is to allow the "Buffer" type to be used properly. Initially when I did this, it tried to import it as ./models/Buffer which is wrong. There is similar work done in the angular client for the "Blob" type.